src/clj/puppetlabs/services/ca/certificate_authority_service.clj

```diff
@@ -9,15 +9,19 @@
             [puppetlabs.services.protocols.ca :refer [CaService]]
             [puppetlabs.comidi :as comidi]
             [puppetlabs.i18n.core :as i18n]
-            [puppetlabs.trapperkeeper.services.status.status-core :as status-core]))
+            [puppetlabs.trapperkeeper.services.status.status-core :as status-core]
+            [puppetlabs.rbac-client.protocols.activity :refer [ActivityReportingService]]
+            [puppetlabs.trapperkeeper.services :refer [maybe-get-service]]
+            [puppetlabs.rbac-client.services.activity :refer [report-activity]]))
```

Comment:
Here you want to pull in https://github.com/puppetlabs/clj-rbac-client/blob/master/src/puppetlabs/rbac_client/protoc/activity.clj#L3 instead and call the protocol function `report-activity!` instead of the helper function in the service file `report-activity`.

Reply:
Here's an example of requiring a protocol: https://github.com/puppetlabs/puppetserver/blob/main/src/clj/puppetlabs/services/master/master_core.clj#L22 and then using it: https://github.com/puppetlabs/puppetserver/blob/main/src/clj/puppetlabs/services/master/master_core.clj#L462-L464